File: Queue/java/GENqueue.java
class node<N>
{
    public N data;
    public node<N> next;
}

class QueueX<N>
{
    private node<N> first;
    private int iCount;

    public QueueX()
    {
        this.first = null;
        this.iCount = 0;
    }

    void enqueue (N iNo)  // insert last
    {
        node<N> newn = new node<N>();

        newn.data = iNo;
        newn.next = null;

        if(first == null)
        {
            first = newn;
        }
        else
        {
            node<N> temp = first;

            while(temp.next != null)
            {
                temp=temp.next;

               
            }
            temp.next = newn;
        }
        iCount++;

    }

    public N dequeue()     // delete first
    {
        node<N> temp = first;
        if(first ==null)
        {
            System.out.println("Queue is empty :");
            return null;
        }
        else
        {
            first = first.next;
            iCount--;
            return temp.data;
        }

       
    }

    void Display()
    {
        node<N> temp = first;
        while(temp != null)
        {
            System.out.print(" | "+temp.data+" | - ");
            temp = temp.next ;
        }
        System.out.println();

    }

    public int Count ()
    {
        return iCount;
    }

}

class   GENqueue
{
    public static void main(String a[])
    {
        QueueX<Integer> sobj = new  QueueX <>();
        int iRet = 0;

        sobj. enqueue (10);
        sobj. enqueue (20);
        sobj. enqueue (30);
        sobj.enqueue(40);

        sobj.Display();
        iRet = sobj.Count();
        System.out.println("element in queue:"+iRet);

        sobj.dequeue();
        
        sobj.Display();
        iRet = sobj.Count();
        System.out.println("element in queue:"+iRet);

        /////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////
        
        QueueX<Double> sobj1 = new  QueueX <>();
        double dRet = 0;

        sobj1. enqueue (10.30);
        sobj1. enqueue (20.70);
        sobj1. enqueue (30.89);
        sobj1.enqueue(40.99);

        sobj1.Display();
        iRet = sobj1.Count();
        System.out.println("element in queue:"+iRet);

        sobj1.dequeue();
        
        sobj1.Display();
        iRet = sobj1.Count();
        System.out.println("element in queue:"+iRet);


        /////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////
        
        QueueX<String> sobj2 = new  QueueX <>();
        String str = null;

        sobj2. enqueue ("c");
        sobj2. enqueue ("cpp");
        sobj2. enqueue ("java");
        sobj2.enqueue("python");

        sobj2.Display();
        iRet = sobj2.Count();
        System.out.println("element in queue:"+iRet);

        sobj2.dequeue();
        
        sobj2.Display();
        iRet = sobj2.Count();
        System.out.println("element in queue:"+iRet);

    }
}  

File: Queue/cpp/GENqueue.cpp
#include<iostream>
using namespace std;

template<class T>
struct node
{
    T data ;
    struct node<T>  *next;
};

template<class T>

class QueueX 
{
    public: struct node<T> *first;
    public: int iCount ;

    public : QueueX()
    {
        this->first = NULL;
        this->iCount = 0;
    }

    void Enqueue(T no)   //insert last
    {
        struct node<T>  * temp = NULL;
        struct node<T>  *newn = NULL;

        newn = new struct node<T> ();

        newn->data = no;
        newn->next = NULL;

        if(first == NULL)
        {
            first = newn;
        }
        else
        {
            temp =first;

            while(temp->next != NULL)
            {
                temp = temp->next;
            }
            temp->next = newn;
        }
        iCount++;
    }

    T Dequeue ()
    {
        struct node<T>  * temp = first;
        T data = 0;

        if(temp == NULL)
        {
            cout<<"Queue is empty no element \n";
            return NULL;
        }
        else
        {
            first = temp->next;
            data = temp->data;
            delete temp;
            iCount--;

            return data;

        }
    }

    void Display()
    {
        struct node<T>  * temp = first;

        while(temp != NULL)
        {
            cout<<" | "<<temp->data<<" | -";
            temp = temp->next;
        }
        cout<<"\n";
    }

    int Count()
    {
        return iCount;
    }

};

int main()
{
    QueueX<int> *sobj = new QueueX<int>();
    int iRet = 0;
    int isize = 0;
    int i = 0;
    int ivalue=0;

    cout<<"enter the how many no of element you want to enter :";
    cin>>isize;

    cout<<"enter the element :\n";
    for(i = 1; i <= isize;i++ )
    {
        cin>>ivalue;
        sobj->Enqueue(ivalue);
    }

    sobj->Display();

    iRet = sobj->Count();
    cout<<"no of element in queue is :"<<iRet<<"\n";

    iRet = sobj->Dequeue();
    cout<<"removed element from  Queue is :"<<iRet<<"\n";

    sobj->Display();

    iRet = sobj->Count();
    cout<<"no of element in queue is :"<<iRet<<"\n";

    delete sobj;

    /////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////

    QueueX<double> *sobj1 = new QueueX<double>();
    double dvalue = 0.0;
    double dRet = 0.0;

    cout<<"enter the how many no of element you want to enter  :";
    cin>>isize;

    cout<<"enter the element in doble format :\n";
    for(i = 1; i <= isize;i++ )
    {
        cin>>dvalue;
        sobj1->Enqueue(dvalue);
    }

    sobj1->Display();

    iRet = sobj1->Count();
    cout<<"no of element in queue is :"<<iRet<<"\n";

    dRet = sobj1->Dequeue();
    cout<<"removed element from  Queue is :"<<dRet<<"\n";

    sobj1->Display();

    iRet = sobj1->Count();
    cout<<"no of element in queue is :"<<iRet<<"\n";
    delete sobj1;

    return 0;
}